src/lib.rs

```diff
@@ -751,6 +751,41 @@ mod tests {
         unsafe { ffi::secp256k1_context_destroy(ctx_full) };
     }
 
+    #[test]
+    fn test_raw_ctx() {
+        let ctx_full = Secp256k1::new();
+        let ctx_sign = Secp256k1::signing_only();
+        let ctx_vrfy = Secp256k1::verification_only();
+
+        let full = unsafe {Secp256k1::from_raw_all(ctx_full.ctx)};
+        let sign = unsafe {Secp256k1::from_raw_signining_only(ctx_sign.ctx)};
+        let vrfy = unsafe {Secp256k1::from_raw_verification_only(ctx_vrfy.ctx)};
+
+        let (sk, pk) = full.generate_keypair(&mut thread_rng());
+        let msg = Message::from_slice(&[2u8; 32]).unwrap();
+        // Try signing
+        assert_eq!(sign.sign(&msg, &sk), full.sign(&msg, &sk));
+        let sig = full.sign(&msg, &sk);
+
+        // Try verifying
+        assert!(vrfy.verify(&msg, &sig, &pk).is_ok());
+        assert!(full.verify(&msg, &sig, &pk).is_ok());
+
+        drop(full);drop(sign);drop(vrfy);
+        drop(ctx_full);drop(ctx_sign);drop(ctx_vrfy);
```

Comment:
Probably not needed. But the manual drops here are meant to try and make sure the `ctx_*` don't drop before the Context's using it.